tests/regression/74-use_after_free/13-multi-threaded-uaf-with-joined-thread.c

```diff
@@ -0,0 +1,33 @@
+//PARAM: --set ana.activated[+] useAfterFree --set ana.activated[+] threadJoins
+#include <stdlib.h>
+#include <stdio.h>
+#include <pthread.h>
+
+int* gptr;
+
+// Mutex to ensure we don't get race warnings, but the UAF warnings we actually care about
+pthread_mutex_t mtx = PTHREAD_MUTEX_INITIALIZER;
+
+void *t_use(void* p) {
+    pthread_mutex_lock(&mtx);
+    *gptr = 0; //NOWARN
```

Comment:
I think the reason we still produce a warning here is the following: We check the set of potentially started threads calling `free`, removing those which will definitely be must-joined at the point when `free` is called.

Here, however it is the **current thread** that will be must joined at the point at which free is called.

That seems like it is somehow a separate issue from checking whether any of the threads calling free can already have been started?

Reply:
Ideally, we'd also want to add a test for the case which your current implementation now catches.

Reply:
> That seems like it is somehow a separate issue from checking whether any of the threads calling free can already have been started?

I think I found a solution for removing the WARN now. Cf. https://github.com/goblint/analyzer/pull/1123/commits/cd95e8d45dbcaf68adacd3508cc5938eec4d500c for the solution idea.

Reply:
> Ideally, we'd also want to add a test for the case which your current implementation now catches.

I believe we already had one covering the implementation before https://github.com/goblint/analyzer/commit/cd95e8d45dbcaf68adacd3508cc5938eec4d500c, i.e., with the "current implementation" from a few weeks ago, namely test case `74/12`.